### ISSUE TITLE
Rendering: partial repaint next to a compositing border

### ISSUE THREAD

**Issue by igor-borovkov** [WebPlatformForEmbedded/WPEWebKit#390]
We observe multiple instances with DIV elements not being repainted fully near a compositing border. There is no simple html / css sample currently available, but there was a reliable scenario discovered recently on Watchable website. 

1. https://watchablex1appserver-prod.xidio.com/channelstore-webapp/#browse/my_shows
2. Click on Login button
2. In a pop up screen navigate down to a second row, to highlight "1 2 3" button

Result:  "1 2 3" button is not fully highlighted (tiny portion is still white)

photo:  https://photos.app.goo.gl/Q26WWVjGrjGzSGOl2

I believe this issue goes away if manually disable accelerated compositing

**Comment by magomez** [comment]
In which version is this happening? I've fixed something similar a while ago. Is this reproducible using current master?

**Comment by igor-borovkov** [comment]
I'll check if we got that patch in, I remember your patch (images and accelerated compositing thing), have to check if it got to your stable, we're switching soon to the latest stable commit, the last update was August 1st.

**Comment by wouterlucas** [comment]
Its in stable, I believe its this one https://github.com/WebPlatformForEmbedded/WPEWebKit/commit/78e5f8d92b714ae5cf1efd552592648610a50d86 added on Oct 4th.

**Comment by igor-borovkov** [comment]
Tried with the latest stable and with our stable (July 5th) plus those two patches:
https://github.com/WebPlatformForEmbedded/WPEWebKit/commit/9a02007ed0010dc8adb5526d20aff7213f8324cb?diff=unified

https://github.com/WebPlatformForEmbedded/WPEWebKit/commit/78e5f8d92b714ae5cf1efd552592648610a50d86

still reproducible for me. I guess some other change might have fixed this in master.

**Comment by magomez** [comment]
I'm not able to reproduce this behavior with the stable branch, neither with current master nor with 753acba767735eacd97518bf22f0ee9f046573f1 which is from July 30, on the rpi2. I'm going to test with a broadcom box.

**Comment by magomez** [comment]
Ok, I'm able to reproduce the issue on the bcm7429 box using the current stable branch, but indeed it's not reproducible with the master one. But I don't think this is related to compositing. It seems more like the clipping rect being used by cairo to render the button is wrong, but just sometimes, as the buttons in the upper row are similar and they are rendered fine.

**Comment by magomez** [comment]
Just an update. I'm still investigating this. I've been comparing sizes and positions between right and wrong renderings and I'm seeing that when the rendering is wrong there seems to be an extra pixel added to the size of the divs. In the case of single characters, for example, the size should always be 30x30 pixels, but the @ div for example, which is rendered wrong, has a size of 31x30. This extra pixel causes a misalignment of the div content and I think that causes that the last column of the div doesn't get rendered properly. I'm still trying to find out where that extra pixel is coming from but it's being tricky.

**Comment by igor-borovkov** [comment]
Thanks for the update, Miguel! I guess there is no master commit looking obvious it could have fixed this issue ?

**Comment by magomez** [comment]
I'm afraid there are hundreds of commits in between master and stable. I need to locate the place in the code where the different sizes show up before being able to look for differences with master and find the fix.

**Comment by magomez** [comment]
It has been a complex debug but I think I found the problem. And the fix is quite simple. I've created a bug upstream (https://bugs.webkit.org/show_bug.cgi?id=179476) cause despite it's not reproducible in the test page there (because the divs don't get decimal sizes there), I think it could happen in other pages. If the patch is accepted upstream, I'll cherry pick it to the metrological repo. If not, I'll just commit the fix to the stable branch.
